### PR TITLE
Add benchmark data for testing non-vectorised paths

### PR DESCRIPTION
**Pull request by Gnbrkm41**
Related: https://github.com/dotnet/coreclr/pull/27798

The first commit is related to the PR mentioned above; I think it could be worth adding this even if the PR above doesn't make it in, in case there could be a similar change.

The second commit is per https://github.com/dotnet/corefx/pull/39173#issuecomment-516728961.

### CODE REVIEW

File: src/benchmarks/micro/corefx/System.Collections/Perf.BitArray.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class Perf_BitArray
         private const int DefaultShiftCount = 17;
         private const bool BooleanValue = true;
 
-        [Params(Utils.DefaultCollectionSize)]
+        [Params(4, Utils.DefaultCollectionSize)]
```
Comment (adamsitnik) [inline src/benchmarks/micro/corefx/System.Collections/Perf.BitArray.cs]: could you please add a comment explaining why we have two test cases? Something similar to:

https://github.com/dotnet/performance/blob/cc73a0126ae9056382fe23a9dee183b18b1b67de/src/benchmarks/micro/corefx/System.Globalization/StringHash.cs#L33-L36

**Comment by Gnbrkm41** [comment]
I'll need to add some more cases to follow up with changed implementations.

**Comment by adamsitnik** [comment]
> I'll need to add some more cases to follow up with changed implementations.

Please tag me when the PR is ready to merge (I would prefer to merge 1 final PR)

**Comment by Gnbrkm41** [comment]
Is the performance repo being merged into runtime or somewhere else as well? Thought this was going to stay.

**Comment by adamsitnik** [comment]
> Is the performance repo being merged into runtime or somewhere else as well?

**Comment by Gnbrkm41** [comment]
Actually... I think this is good as-is. I was going to add a slightly longer string, thinking it would matter since I changed the implementation to process more at once; but it is covered by the current data as well.

**Pull request merged**